filecache/file_cache_path.py

```diff
@@ -0,0 +1,1252 @@
+                is the name of the bucket or webserver or the empty string for a local
+                file, `path` is the rest of the URL, `cache_dir` is the top-level
+                directory of the cache (``<cache_dir>/<cache_name>``), and `cache_subdir`
+                is the subdirectory specific to this scheme and remote. If the translator
+                wants to override the default translation, it can return a Path.
+                Otherwise, it returns None. If the returned Path is relative, if will be
+                appended to `cache_dir`; if it is absolute, it will be used directly (be
+                very careful with this, as it has the ability to access files outside of
+                the cache directory). If more than one translator is specified, they are
+                called in order until one returns a Path, or it falls through to the
+                default.
+
+                If None, use the default value given when this :class:`FCPath` was
+                created.
+        Returns:
+            The Path of the filename in the temporary directory (or the original absolute
+            path if local). If `sub_path` was a list or tuple of paths, then instead
+            return a list of Paths of the filenames in the temporary directory (or the
+            original absolute path if local). If `exception_on_fail` is False, any Path
+            may be an Exception if that file does not exist or the upload failed.
+
+        Raises:
+            FileNotFoundError: If a file to upload does not exist or the upload failed,
+                and exception_on_fail is True.
+            ValueError: If the derived path is not absolute.
+        """
+
+        old_upload_counter = self._filecache_to_use.upload_counter
+
+        nthreads = self._validate_nthreads(nthreads)
+
+        try:
+            if isinstance(sub_path, (list, tuple)):
+                new_sub_paths = [FCPath._join(self._path, p) for p in sub_path]
+                if not all([FCPath._is_absolute(x) for x in new_sub_paths]):
+                    raise ValueError(
+                        f'Derived paths must be absolute, got {new_sub_paths}')
+                ret = self._filecache_to_use.upload(cast(StrOrPathOrSeqType,
+                                                         new_sub_paths),
+                                                    anonymous=self._anonymous,
+                                                    nthreads=nthreads,
+                                                    exception_on_fail=exception_on_fail,
+                                                    url_to_path=url_to_path)
+            else:
+                new_sub_path = FCPath._join(self._path, sub_path)
+                if not FCPath._is_absolute(str(new_sub_path)):
+                    raise ValueError(
+                        'Derived path must be absolute, got {new_sub_path}')
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Fix f-string formatting in ValueError message</b></div>
<div id="fix">

The f-string in the `ValueError` message is not properly formatted. The variable `new_sub_path` should be enclosed in curly braces. A similar issue was also found in filecache/file_cache_path.py (line 610-612).
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -889 +889 @@
 -                        'Derived path must be absolute, got {new_sub_path}')
 +                        f'Derived path must be absolute, got {new_sub_path}')
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/SETI/rms-filecache/pull/24#issuecomment-2495681924>#ad05ee</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged

Reply:
Good suggestion. Done.